substrate/client/chain-spec/src/genesis_block.rs

```diff
@@ -43,12 +43,7 @@ where
 		let runtime_code = sp_core::traits::RuntimeCode {
 			code_fetcher: &code_fetcher,
 			heap_pages: None,
-			hash: {
-				use std::hash::{Hash, Hasher};
-				let mut state = DefaultHasher::new();
-				wasm.hash(&mut state);
-				state.finish().to_le_bytes().to_vec()
-			},
+			hash: sp_core::blake2_256(wasm).to_vec(),
```

Comment:
This is not correct, you need to get the hasher used by the `Block` type. For most chains this is blake256, but not for all.

Reply:
I mean "not correct" not being really the right term here, but if we want to fix this, we should fix it properly.

Reply:
https://github.com/paritytech/polkadot-sdk/blob/36b7037c6c0a587c11f9f096b2893a5755dd64ce/substrate/primitives/state-machine/src/backend.rs#L427-L433

Reply:
Change the function to take `H: Hash` and then you can pass `HashingOf<Block>` on the calling side of this function.

Reply:
Thanks, it was easier than I expected.